### ISSUE TITLE
Disallow punctuation at the end of in list/tag names

### ISSUE THREAD

**Issue by roman-yagodin** [mpcjanssen/simpletask-android#1077]
Then editing tasks manually, I often use list/tag (context/project) labels as parts of the speech, which sometimes produces duplicating labels in Simpletask, like `+Project` and `+Project,`

I think the way how Simpletask recognizes lists and tags should be adjusted to allow lists/tag to be separated not only by whitespace, but also by punctuation (special chars) followed by whitespace.  In other words, last label char should always be alphanumeric.

**Comment by mpcjanssen** [comment]
This was a result of allowing punctuation inside tags. Having punctuation at the end of lists or tags seems much less useful. A configurable regexp sounds like a bit of overkill though. Maybe a checkbox in the settings.

**Comment by smichel17** [comment]
What about just fixing this one case? Allow any punctuation within tags, but strip a subset of trailing punctuation, specifically any of these symbols: `.,:;?!')]}>`

Without any preference, I think that would probably cover 95% of use cases.

Or -- I'd need to check how this one interacts with the todo.txt spec -- if you want trailing punctuation, use the same symbol as a closing tag. Eg, if you wanted the tag `Project?`, you'd write it `+Project?+`.

**Comment by mpcjanssen** [comment]
spec is silent on these kinds of things

**Comment by roman-yagodin** [comment]
I agree with @smichel17 - it would be good to preserve ability to use not only alphanumeric chars at the end of the label. So there is some options:

- Special ending char like `+Project?_`
- Escape char like `+Project\?`
- Some kind of brackets like `+(Project?)`
